1.segment_images/scripts/1.segment_cells_organoids.py

```diff
@@ -0,0 +1,291 @@
+
+# Import dependencies
+import numpy as np
+import skimage
+import tifffile
+from cellpose import core, io, models
+
+io.logger_setup()
+from cellpose.io import imread
+from PIL import Image
+from skimage import io
+
+use_GPU = core.use_gpu()
+print(">>> GPU activated? %d" % use_GPU)
+
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+print(in_notebook)
+
+
+# In[2]:
+
+
+if not in_notebook:
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+    parser.add_argument(
+        "--window_size", type=int, help="Size of the window to use for the segmentation"
+    )
+    parser.add_argument(
+        "--clip_limit",
+        type=float,
+        help="Clip limit for the adaptive histogram equalization",
+    )
+
+    args = parser.parse_args()
+    window_size = args.window_size
+    clip_limit = args.clip_limit
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+
+else:
+    input_dir = pathlib.Path("../examples/raw_z_input/").resolve(strict=True)
+    window_size = 3
+    clip_limit = 0.1
+
+mask_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
+mask_path.mkdir(exist_ok=True, parents=True)
+
+
+# ## Set up images, paths and functions
+
+# In[3]:
+
+
+image_extensions = {".tif", ".tiff"}
+files = sorted(input_dir.glob("*"))
+files = [str(x) for x in files if x.suffix in image_extensions]
+
+
+# In[4]:
+
+
+for f in files:
+    print(f)
+    if "405" in f:
+        nuclei = io.imread(f)
+    elif "488" in f:
+        cyto1 = io.imread(f)
+    elif "555" in f:
+        cyto2 = io.imread(f)
+    elif "640" in f:
+        cyto3 = io.imread(f)
+    else:
+        print(f"Unknown channel: {f}")
+
+# pick which channels to use for cellpose
+cyto = skimage.exposure.equalize_adapthist(cyto2, clip_limit=clip_limit)
+
+
+original_nuclei_image = nuclei.copy()
+original_cyto_image = cyto.copy()
+
+original_nuclei_z_count = nuclei.shape[0]
+original_cyto_z_count = cyto.shape[0]
+
+
+# In[5]:
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
+image_stack_2_5D = np.empty((0, cyto.shape[1], cyto.shape[2]), dtype=cyto.dtype)
+for image_index in range(cyto.shape[0]):
+    image_stack_window = cyto[image_index : image_index + window_size]
+    if not image_stack_window.shape[0] == window_size:
+        break
+    # max project the image stack
+    image_stack_2_5D = np.append(
+        image_stack_2_5D, np.max(image_stack_window, axis=0)[np.newaxis, :, :], axis=0
+    )
+
+image_stack_2_5D = np.array(image_stack_2_5D)
+cyto = np.array(image_stack_2_5D)
+print("2.5D cyto image stack shape:", cyto.shape)
+
+
+# make a 2.5 D max projection image stack with a sliding window of 3 slices
```

Comment:
Similar to the previous comment, this pertains specifically to the nuclei-related code. It appears to be duplicated—could this be refactored into a utility function for better reusability? If refactoring impacts downstream functionality, feel free to leave it as is.